addon/components/inputs/when.js

```diff
@@ -0,0 +1,201 @@
+  },
+
+  // == Computed Properties ====================================================
+
+  @readOnly
+  @computed()
+  /**
+   * A unique ID to be used in order to find the date-picker to enable/disable
+   * @returns {String} unique ID for the date-time-picker
+   */
+  dateId () {
+    return guidFor(this) + '_dateId'
+  },
+
+  @readOnly
+  @computed('cellConfig')
+  /**
+   * The date format to be used in the date-time-picker
+   * @param {Object} cellConfig - contains the dateFormat "string" property (a valid moment format)
+   * @see {@link https://momentjs.com/docs/#/displaying/format}
+   * @returns {String} the date format
+   */
+  dateFormat (cellConfig) {
+    return get(cellConfig, 'renderer.dateFormat') || 'YYYY-MM-DD'
+  },
+
+  @readOnly
+  @computed('cellConfig')
+  /**
+   * The time format to be used in the date-time-picker
+   * @param {Object} cellConfig - contains the timeFormat "string" property (a valid moment format)
+   * @see {@link https://momentjs.com/docs/#/displaying/format}
+   * @returns {String} the time format
+   */
+  timeFormat (cellConfig) {
+    return get(cellConfig, 'renderer.timeFormat') || 'HH:mm:ss'
+  },
+
+  @readOnly
+  @computed('cellConfig')
+  /**
+   * The date time format to be used for formating the value sent to the bunsen model
+   * @param {Object} cellConfig - contains the dateTimeFormat "string" property (a valid moment format)
+   * @see {@link https://momentjs.com/docs/#/displaying/format}
+   * @returns {String} the date time format
+   */
+  dateTimeFormat (cellConfig) {
+    return get(cellConfig, 'renderer.dateTimeFormat') || 'YYYY-MM-DDTHH:mm:ssZ'
+  },
+
+  @readOnly
+  @computed()
+  /**
+   * The unique "Id" for the radio group
+   * @returns {String} the unique id
+   */
+  groupId () {
+    return guidFor(this) + '_radioGroup'
+  },
+
+  @readOnly
+  @computed('cellConfig')
+  /**
+   * The size to be used for the radio buttons
+   * @param {Object} cellConfig - contains the size "string" property (a valid frost-radio-button size)
+   * @returns {String} the radio button size
+   */
+  size (cellConfig) {
+    return get(cellConfig, 'renderer.size') || 'small'
+  },
+
+  // == Functions ==============================================================
+
+  /**
+   * Initializes the component properties, sets the first radio button as selected and
+   * sets the bunsen model to the value of the first radio button
+   * @returns {undefined}
+   */
+  init () {
+    this._super(...arguments)
+
+    const currentDateTime = moment()
+    const date = moment(currentDateTime).format(this.get('dateFormat'))
+    const time = moment(currentDateTime).format(this.get('timeFormat'))
+    const firstButtonValue = get(this, 'cellConfig.renderer.value')
+
+    this.setProperties({
+      date,
+      firstButtonValue,
+      selectedValue: firstButtonValue,
+      storedDateTimeValue: moment(currentDateTime).format(this.get('dateTimeFormat')),
+      time
+    })
+
+    run.later(() => {
```

Comment:
I'm not sure it is necessary either, pretty sure doing a `setProperties` in `init` does not trigger a re-render anyway, but there might be some reason I'm not thinking of to exhaust the runLoop before calling the `onChange` here, hence the `run.later`